src/server/middlewares/validateVote.js

```diff
@@ -1,175 +1,260 @@
 import httpStatus from 'http-status';
 
-import { isSignatureValid } from '~/server/services/crypto';
-import { packVote, packBooth } from '~/common/services/encoding';
+import APIError from '~/server/helpers/errors';
+import Answer from '~/server/models/answer';
+import Question from '~/server/models/question';
+import Vote from '~/server/models/vote';
+import logger from '~/server/helpers/logger';
 import {
   getQuestionContract,
   getAdminContract,
 } from '~/common/services/contracts';
 import { isActiveFestival } from '~/common/services/contracts/festivals';
-import Vote from '~/server/models/vote';
-import Question from '~/server/models/question';
-import APIError from '~/server/helpers/errors';
+import { isSignatureValid } from '~/server/services/crypto';
+import { packVote, packBooth } from '~/common/services/encoding';
 
-const admin = getAdminContract(process.env.ADMIN_CONTRACT);
+const adminContract = getAdminContract(process.env.ADMIN_CONTRACT);
 
-const checkBooth = async (vote, question) => {
-  const booth = await admin.methods.getVotingBooth(vote.booth).call();
-  const festival = await question.methods.festival().call();
-  vote.festival = festival;
-  if (!booth[0] || festival !== booth[2]) {
-    throw Error('invalid booth');
+async function checkBooth(vote, questionContract) {
+  const result = await adminContract.methods
+    .getVotingBooth(vote.boothAddress)
+    .call();
+  const isInitialized = result[0];
+  const isDeactivated = result[1];
+  const festivalId = result[2];
+
+  const festivalChainId = await questionContract.methods.festival().call();
+
+  // Check if ..
+  // 1. Festival is active
```

Comment:
doesn't seem to be checking if festival is active here, that happens down in checkFestival